leagues.go

```diff
@@ -348,44 +348,34 @@ func (app *App) RegisterLeague(w http.ResponseWriter, r *http.Request) {
 	userID := r.Context().Value("user_id").(int)
 
 	// Get the capacity and registered count from the leagues table
+	// Order the struct fields in an optimal way to avoid padding
 	var result struct {
-		Capacity        int
-		Registered      int
-		RegisteredUsers string
-	}
-	err := app.DB.Raw("SELECT capacity, registered, users_registered FROM leagues WHERE league_id = ?", leagueID).Scan(&result).Error
-	capacity := result.Capacity
-	registered := result.Registered
+		RegisteredUsers string    `gorm:"column:users_registered"`
+		leagueStatus    string    `gorm:"column:league_status"`
+		createdAt       time.Time `gorm:"column:created_at"`
+		Capacity        int       `gorm:"column:capacity"`
+		Registered      int       `gorm:"column:registered"`
+		waitTime        int       `gorm:"column:wait_time"`
+	}
+	err := app.DB.Raw("SELECT capacity, registered, users_registered, league_status, created_at, wait_time FROM leagues WHERE league_id = ?", leagueID).Scan(&result).Error
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	capacity := result.Capacity
+	registered := result.Registered
 
-	// Check if the league is full
-	if registered == capacity {
-		http.Error(w, "League is full", http.StatusBadRequest)
+	// Check if the league is full and/or if the leage is no longer accepting registrations
+	if registered == capacity || result.leagueStatus != "waiting" || time.Now().Sub(result.createdAt).Seconds() > float64(result.waitTime) {
+		http.Error(w, "League has started, cannot join now", http.StatusBadRequest)
 		return
 	}
 
-	// Extract the comma-separated values in result.RegisteredUsers into a list
-	var registeredUsers []string
-	if result.RegisteredUsers == "" {
-		registeredUsers = []string{}
-	} else {
-		registeredUsers = strings.Split(result.RegisteredUsers, ",")
-	}
-	// append the new user to the list
-	registeredUsers = append(registeredUsers, fmt.Sprintf("%d", userID))
-
-	// convert the list back to a comma-separated string
-	var newRegisteredUsers string
-	if len(registeredUsers) == 1 {
-		newRegisteredUsers = registeredUsers[0]
-	} else {
-		newRegisteredUsers = strings.Join(registeredUsers, ",")
-	}
+	// Add the user to the users_registered list
+	newRegisteredUsers := result.RegisteredUsers + fmt.Sprintf(",%d", userID)
```

Comment:
When the first user comes and register to the league this string newRegisteresUsers looks something like : ",1"
Handle this case.